results/cmd/api/main_test.go

```diff
@@ -114,14 +115,15 @@ func TestUpdateTaskRun(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 	defer cancel()
 	objectmeta := &pb.ObjectMeta{Uid: "123459", Name: "mytaskrun", Namespace: "default"}
-	r, err := srv.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "1", Metadata: objectmeta}})
+	r, err := srv.CreateTaskRunResult(ctx, &pb.CreateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "v1beta1", Metadata: objectmeta}})
 	if err != nil {
 		t.Fatalf("could not create taskrun: %v", err)
 	}
 
 	// Update the created taskrun
 	objectmeta = &pb.ObjectMeta{Uid: "123459", Name: "ziweifan", Namespace: "tomorrow"}
-	r, err = srv.UpdateTaskRunResult(ctx, &pb.UpdateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "v1alpha1", Metadata: objectmeta}, ResultsId: r.GetResultsId()})
+	fmask, _ := fieldmask.New(&pb.TaskRun{}, "api_version", "metadata.uid")
+	r, err = srv.UpdateTaskRunResult(ctx, &pb.UpdateTaskRunRequest{TaskRun: &pb.TaskRun{ApiVersion: "v1alpha1", Metadata: objectmeta}, ResultsId: r.GetResultsId(), FieldMask: fmask})
```

Comment:
hmmmm - I'm not sure what's going on here. 

The update mask includes uid and api_version, but we're modifying name. If this is passing, this means that name is still being modified even though it's not included in the mask.

We should verify this is working as intended.

Reply:
Perhaps this is due to the FromFieldMask parameters?

Reply:
This is the same as what I intend to do in the last comment - updating only field in fieldmask

Reply:
ahhh. I assumed that the only thing that was being changed was the objectmeta. It wasn't clear to me that other things were being changed as well, because they were embedded in the Update call. We should break out any field updates to make it clearer which fields are being changed.

I suspect the table driven tests will also help make this clearer.